libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -702,29 +713,39 @@ void AP_GPS::detect_instance(uint8_t instance)
             new_gps = new AP_GPS_UBLOX(*this, state[instance], _port[instance], role);
         }
 #ifndef HAL_BUILD_AP_PERIPH
+#if AP_GPS_SBP2_ENABLED
         else if ((_type[instance] == GPS_TYPE_AUTO || _type[instance] == GPS_TYPE_SBP) &&
                  AP_GPS_SBP2::_detect(dstate->sbp2_detect_state, data)) {
             new_gps = new AP_GPS_SBP2(*this, state[instance], _port[instance]);
         }
+#endif //AP_GPS_SBP2_ENABLED
+#if AP_GPS_SBP_ENABLED
         else if ((_type[instance] == GPS_TYPE_AUTO || _type[instance] == GPS_TYPE_SBP) &&
                  AP_GPS_SBP::_detect(dstate->sbp_detect_state, data)) {
             new_gps = new AP_GPS_SBP(*this, state[instance], _port[instance]);
         }
-#if !HAL_MINIMIZE_FEATURES
+#endif //AP_GPS_SBP_ENABLED
+#if !HAL_MINIMIZE_FEATURES && AP_GPS_SIRF_ENABLED
         else if ((_type[instance] == GPS_TYPE_AUTO || _type[instance] == GPS_TYPE_SIRF) &&
                  AP_GPS_SIRF::_detect(dstate->sirf_detect_state, data)) {
             new_gps = new AP_GPS_SIRF(*this, state[instance], _port[instance]);
         }
 #endif
+#if AP_GPS_ERB_ENABLED
         else if ((_type[instance] == GPS_TYPE_AUTO || _type[instance] == GPS_TYPE_ERB) &&
                  AP_GPS_ERB::_detect(dstate->erb_detect_state, data)) {
             new_gps = new AP_GPS_ERB(*this, state[instance], _port[instance]);
-        } else if ((_type[instance] == GPS_TYPE_NMEA ||
+        }
+#endif // AP_GPS_ERB_ENABLED
+#if AP_GPS_NEMA_ENABLED
```

Comment:
I wonder of the utility of adding `AP_GPS_HEMI_ENABLED` in the future; I guess it won't save much space!

Reply:
very little...same with ALLYSTAR